clang/test/Sema/warn-thread-safety-analysis.c

```diff
@@ -29,6 +30,22 @@ struct LOCKABLE Mutex {};
 
 struct Foo {
   struct Mutex *mu_;
+  int  a_value GUARDED_BY(mu_);
+
+  struct Bar {
+    struct Mutex *other_mu ACQUIRED_AFTER(mu_);
```

Comment:
I'm not entirely sure how nested `struct`s work in C, but should we be able to reference the parent from the child? At least in C++, you can declare objects of the child independent of the parent, so this wouldn't be meaningful.

This is an existing issue with with the `ACQUIRED_{AFTER,BEFORE}` implementation, but if you add a test for this (which is Ok to make sure it doesn't crash), you should probably also add a comment that this should perhaps be disallowed because it's not meaningful (as I understand).

Reply:
> I'm not entirely sure how nested structs work in C, but should we be able to reference the parent from the child?

I think so, it does read nicely it feels – I see why this could cause a problem though: if the child structure is used outside of the parent, there is an ambiguity. Maybe this should be restricted to anonymous struct? Or maybe we need to taint the child struct so that it's not reused outside of the parent?

Anyhow, I'll add a comment as suggested!

Reply:
@pdherbemont Did you add the comment? I can't find it in the diff shown on GitHub